e2e/e2e_test.go

```diff
@@ -0,0 +1,213 @@
+							"--cloud-provider":                    "azure",
+							"--cluster-dns":                       "10.0.0.10",
+							"--cluster-domain":                    "cluster.local",
+							"--dynamic-config-dir":                "/var/lib/kubelet",
+							"--enforce-node-allocatable":          "pods",
+							"--event-qps":                         "0",
+							"--eviction-hard":                     "memory.available<750Mi,nodefs.available<10%,nodefs.inodesFree<5%",
+							"--feature-gates":                     "RotateKubeletServerCertificate=true",
+							"--image-gc-high-threshold":           "85",
+							"--image-gc-low-threshold":            "80",
+							"--image-pull-progress-deadline":      "30m",
+							"--keep-terminated-pod-volumes":       "false",
+							"--kube-reserved":                     "cpu=100m,memory=1638Mi",
+							"--kubeconfig":                        "/var/lib/kubelet/kubeconfig",
+							"--max-pods":                          "110",
+							"--network-plugin":                    "kubenet",
+							"--node-status-update-frequency":      "10s",
+							"--non-masquerade-cidr":               "0.0.0.0/0",
+							"--pod-infra-container-image":         "mcr.microsoft.com/oss/kubernetes/pause:3.5",
+							"--pod-manifest-path":                 "/etc/kubernetes/manifests",
+							"--pod-max-pids":                      "-1",
+							"--protect-kernel-defaults":           "true",
+							"--read-only-port":                    "0",
+							"--resolv-conf":                       "/run/systemd/resolve/resolv.conf",
+							"--rotate-certificates":               "false",
+							"--streaming-connection-idle-timeout": "4h",
+							"--tls-cert-file":                     "/etc/kubernetes/certs/kubeletserver.crt",
+							"--tls-cipher-suites":                 "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_128_GCM_SHA256",
+							"--tls-private-key-file":              "/etc/kubernetes/certs/kubeletserver.key",
+						},
+					},
+					Distro: datamodel.AKSUbuntuContainerd1804Gen2,
+				},
+			},
+			LinuxProfile: &datamodel.LinuxProfile{
+				AdminUsername: "azureuser",
+			},
+			ServicePrincipalProfile: &datamodel.ServicePrincipalProfile{
+				ClientID: values.AadClientId,
+				Secret:   values.AadClientSecret,
+			},
+
+			CertificateProfile: &datamodel.CertificateProfile{
+				CaCertificate:        string(caCertDecoded),
+				ClientCertificate:    string(clientCertDecoded),
+				APIServerCertificate: string(apiServerCertDecoded),
+				ClientPrivateKey:     string(clientKeyDecoded),
+			},
+		},
+	}
+
+	//Adding a dummy key because we are not actually ssh'ing into the node. 
+	cs.Properties.LinuxProfile.SSH.PublicKeys = []datamodel.PublicKey{{
```

Comment:
the SSH profile can be left empty if not needed?

Reply:
we need the AdminUsername for the cseCmd which is populated through the SSh profile. And if the AdminUsername is define, the logic requires SSH key to be present as well. So I had to insert a dummy value. However, I can try applying the cseCmd without the AdminUsername.